### ISSUE TITLE
Entropic open-set loss using MEAN rather than SUM over the outputs

### ISSUE THREAD

**Issue by rafaelvareto** [Vastlab/vast#54]
I'm afraid the PyTorch implementation of the Entropic Open-set Loss is not matching Equation 1 from the papers Watchlist Adaptation and Reducing Network Agnostophobia. In both papers, it indicates the using of the array operation SUM instead of MEAN and I believe it ends up having an unexpected result.

https://github.com/Vastlab/vast/blob/fd7dfa046d15c501276ef7b4439bbd757e4b2055/vast/losses/losses.py#L136

It also ends up impacting the chosen parameter for lambda, when combined with the Objectosphere Loss, especially with a large number of classes.

**Comment by siebenkopf** [comment]
@akshay-raj-dhamija I second this change. The `mean` above is on the level of outputs, which is wrong. For a one-hot vector (known sample), the total loss will be `1/C * log(S_c)`, but it should be `log(S_c)`. Similarly, for unknown samples, this would result in `1/C \sum_c 1/C log(S_c)`, which is not correct.

In theory, this just modifies the learning rate by a factor of `1/C`. But now, when combining entropic with Objectosphere, only the former is scaled by `1/C`, and not the latter. Especially for large number of classes, finding a good balance between entopic and Objectosphere becomes more difficult.

**Comment by akshay-raj-dhamija** [comment]
Agreed! Thanks, @rafaelvareto for pointing this out.
